### PR TITLE
remove 2 deprecated create_device_proto_* methods for v0.16 release

### PR DESCRIPTION
**Pull request by kmlau**
This is the 1st task under "Remove deprecations" section in https://github.com/quantumlib/Cirq/issues/5050

### CODE REVIEW

**Review by dstrain115 — APPROVED** [review]

File: cirq-google/cirq_google/devices/grid_device_test.py
Comment (verult) [inline cirq-google/cirq_google/devices/grid_device_test.py:421]: This test can be removed, since we no longer need to be backward compatible with SerializableGateSet

**Review by verult — APPROVED** [review]
LGTM. I'm chatting with @maffoo to verify that it's OK to merge removal PRs before server-side changes are in for Cirq 0.15 deprecations. Until then let's hold off on merging

File: cirq-google/cirq_google/devices/grid_device_test.py
Comment (kmlau) [reply cirq-google/cirq_google/devices/grid_device_test.py]: done

**Review by maffoo — APPROVED** [review]

**Comment by verult** [comment]
Synced with @maffoo - we'll merge all the removals before updating Pyle.

**Pull request merged**
